stacker/plan.py

```diff
@@ -182,14 +191,30 @@ def _single_run(self):
                 )
                 continue
 
+            if not step.done and self._watch_func and step_name not in self._watchers:
```

Comment:
added a test in:  a089dc3

i've also been using this for a couple of months now